src/main/java/it/gov/pagopa/pu/debtpositions/service/InstallmentServiceImpl.java

```diff
@@ -73,4 +87,52 @@ public String updateInstallmentNotificationDate(UpdateInstallmentNotificationDat
     return debtPositionUpdateInstallmentService.updateInstallment(debtPositionDTO, updatedInstallments, wfExecutionParameters, accessToken, operatorExternalUserId).getRight();
   }
 
+  @Override
+  public InstallmentDTO updateInstallmentNotificationFee(Long organizationId, String nav, List<DebtPositionOrigin> debtPositionOrigin, long notificationFeeCents,
+    WfExecutionParameters wfExecutionParameters, String accessToken, String operatorExternalUserId) {
+    List<InstallmentDTO> installments = installmentPIIRepository.getByOrganizationIdAndNav(organizationId, nav, debtPositionOrigin).stream()
+      .filter(installment -> InstallmentUtils.MODIFIABLE_STATUSES.contains(installment.getStatus()))
+      .map(installmentMapper::mapToDto)
+      .toList();
+
+    if(installments.isEmpty())
+      throw new NotFoundException("The installment with NAV: "+nav+" was not found");
+    if(installments.size() > 1)
+      throw new ConflictErrorException("Found more than one installment processable with NAV: "+nav);
+
+    InstallmentDTO installment = calculateNewAmount(installments.getFirst(), notificationFeeCents);
+
+    DebtPosition debtPosition = debtPositionRepository.findByInstallmentId(installment.getInstallmentId());
+    debtPositionUpdateInstallmentService.updateInstallment(debtPositionMapper.mapToDto(debtPosition), Collections.singletonList(installment),
```

Comment:
this service is expecting that the `installments2operate` is already applied on the DebtPositionDTO.
Thus, you should find and replace the original Installment with the updated one (or you should search the debtPositionDTO given orgId and nav, and then filter and operate directly on it's installment
(see updateInstallmentNotificationDate method)